app/controllers/UploadSupportingMaterialMultipleController.scala

```diff
@@ -16,61 +16,71 @@
-                                        override val messagesApi: MessagesApi,
-                                        dataCacheConnector: DataCacheConnector,
-                                        navigator: Navigator,
-                                        identify: IdentifierAction,
-                                        getData: DataRetrievalAction,
-                                        requireData: DataRequiredAction,
-                                        formProvider: UploadSupportingMaterialMultipleFormProvider
-                                      ) extends FrontendController with I18nSupport {
+                                                            appConfig: FrontendAppConfig,
+                                                            override val messagesApi: MessagesApi,
+                                                            dataCacheConnector: DataCacheConnector,
+                                                            navigator: Navigator,
+                                                            identify: IdentifierAction,
+                                                            getData: DataRetrievalAction,
+                                                            requireData: DataRequiredAction,
+                                                            formProvider: UploadSupportingMaterialMultipleFormProvider,
+                                                            fileService: FileService
+                                                          ) extends FrontendController with I18nSupport {
 
   val form = formProvider()
 
   def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
     implicit request =>
 
-      val preparedForm = request.userAnswers.get(UploadSupportingMaterialMultiplePage) match {
-        case None => form
-        case Some(value) => form.fill(value)
-      }
+      val existingFiles = request.userAnswers.get(UploadSupportingMaterialMultiplePage).getOrElse(Seq.empty)
 
-      Ok(uploadSupportingMaterialMultiple(appConfig, preparedForm, mode))
+      Ok(uploadSupportingMaterialMultiple(appConfig, form, existingFiles, mode))
   }
 
-  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+
+  def onSubmit(mode: Mode): Action[MultipartFormData[TemporaryFile]] = (identify andThen getData andThen requireData).async(parse.multipartFormData) {
     implicit request =>
 
-      form.bindFromRequest().fold(
-        (formWithErrors: Form[_]) =>
-          Future.successful(BadRequest(uploadSupportingMaterialMultiple(appConfig, formWithErrors, mode))),
-        value => {
-          val updatedAnswers = request.userAnswers.set(UploadSupportingMaterialMultiplePage, value)
+      val files: Seq[MultipartFormData.FilePart[Files.TemporaryFile]] = request.body.files.filter(!_.filename.isEmpty)
+
+      Future
+        .sequence(
+          files.map(f => fileService.upload(f))
```

Comment:
use `underscore` and make it more concise

Reply:
Done